tests/NATS.Client.Platform.Windows.Tests/TlsTests.cs

```diff
@@ -0,0 +1,150 @@
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using NATS.Client.Core;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.X509;
+using Xunit.Abstractions;
+using X509Certificate = System.Security.Cryptography.X509Certificates.X509Certificate;
+
+namespace NATS.Client.Platform.Windows.Tests;
+
+public class TlsTests(ITestOutputHelper output)
+{
+    [Fact]
+    public async Task Tls_fails_without_certificates()
+    {
+        await using var server = await NatsServerProcess.StartAsync(config: "resources/configs/tls.conf");
+        await using var nats = new NatsConnection(new NatsOpts { Url = server.Url });
+
+        var exception = await Assert.ThrowsAsync<NatsException>(async () => await nats.ConnectAsync());
+        Assert.Contains("TLS authentication failed", exception.InnerException?.Message);
+        Assert.IsType<AuthenticationException>(exception.InnerException?.InnerException);
+    }
+
+    [Fact]
+    public async Task Tls_with_certificates()
+    {
+        const string caCertPem = "resources/certs/ca-cert.pem";
+        const string clientCertPem = "resources/certs/client-cert.pem";
+        const string clientKeyPem = "resources/certs/client-key.pem";
+
+        var tlsOpts = new NatsTlsOpts
+        {
+            RemoteCertificateValidationCallback = (_, certificate, _, errors) => ValidateCert(errors, certificate, LoadCaCerts(caCertPem)),
+            LoadClientCerts = () =>
+            {
+                var collection = LoadClientCerts(clientCertPem, clientKeyPem);
+                return new ValueTask<X509Certificate2Collection>(collection);
+            },
+        };
+
+        await using var server = await NatsServerProcess.StartAsync(config: "resources/configs/tls.conf");
+        await using var nats = new NatsConnection(new NatsOpts { Url = server.Url, TlsOpts = tlsOpts });
+
+        await nats.PingAsync();
+
+        await using var sub = await nats.SubscribeCoreAsync<int>("foo");
+        for (var i = 0; i < 64; i++)
+        {
+            await nats.PublishAsync("foo", i);
+            Assert.Equal(i, (await sub.Msgs.ReadAsync()).Data);
+        }
+    }
+
+    private static bool ValidateCert(SslPolicyErrors errors, X509Certificate certificate, List<X509Certificate2> trustedCaCerts)
+    {
+        if (errors == SslPolicyErrors.None)
+            return true;
+
+        var cert2 = new X509Certificate2(certificate);
+
+        foreach (var caCert in trustedCaCerts)
+        {
+            var chain = new X509Chain();
```

Comment:
I know they're tests but X509 is _weird_ so I get paranoid about files/etc leaking from tests...

`X509Chain`  implements `IDisposable`, should this be in a using statement for safety?

Reply:
done